cli/popper/commands/cmd_add.py

```diff
@@ -47,21 +44,30 @@ def cli(ctx, pipeline, folder, branch):
 
     owner, repo, pipe_name = pipeline.split('/')
 
+    new_pipe_name, folder = pu.get_name_and_path_for_new_pipeline(
+        folder, pipe_name)
+
     config = pu.read_config()
 
-    if pipe_name in config['pipelines']:
-        pu.fail("Pipeline {} already in repo.".format(pipe_name))
+    if new_pipe_name in config['pipelines']:
+        pu.fail("Pipeline {} already in repo.".format(new_pipe_name))
 
     project_root = pu.get_project_root()
+
     pipelines_dir = os.path.join(project_root, folder)
 
     if not os.path.exists(pipelines_dir):
-        os.mkdir(pipelines_dir)
+        try:
+            os.makedirs(pipelines_dir)
+        except (OSError, IOError) as e:
+            pu.fail("Could not create the necessary path.\n")
 
     gh_url = 'https://github.com/{}/{}/'.format(owner, repo)
     gh_url += 'archive/{}.tar.gz'.format(branch)
 
-    pu.info("Downloading pipeline {}... ".format(pipe_name))
+    pu.info(
+        "Downloading pipeline {} as {}...".format(pipe_name, new_pipe_name)
```

Comment:
maybe say 'downloading pipeline {} to {}` instead as `as`

Reply:
Since I am using pipeline names for both the placeholders, that's why it says `as`. For example, if you are downloading pipeline1 but named it as pipeline2, it'll say `Downloading pipeline pipeline1 as pipeline2...` which seems right, imo? pipeline2 is the name, not the path. What do you say? @ivotron

Reply:
sure, that's ok